### PR TITLE
Autoplay refactor

### LINKED ISSUE #8847: Autoplay refactor
slack discussion
```
- Drop the shield setting
- Keep the notification bar
- Add the preloaded sites for youtube, twitch, etc
- Add an always always option like we have for fullscreen
```

### PR DESCRIPTION
**Pull request by darkdh**
1. Move autoplay to site settings
2. Introduce built-in site settings

fix #8847

Auditors: @bbondy, @bsclifton

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Test Plan:
Covered by webdriver test and unit test

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run indivually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)

### CODE REVIEW

**Comment by darkdh** [comment]
working on test

**Comment by darkdh** [comment]
Please provide built-in site settings list to support ` js/data/siteSettingsList.js`

File: app/browser/reducers/autoplayReducer.js [hunk 1]
```diff
@@ -43,20 +49,20 @@ const showAutoplayMessageBox = (tabId) => {
...
       if (buttonIndex === 0) {
-        appActions.changeSiteSetting(ruleKey, 'noAutoplay', false)
-
+        appActions.changeSiteSetting(origin, 'autoplay', true)
         if (tab && !tab.isDestroyed()) {
           tab.reload()
+          tab.on('destroyed', function temporaryAllow (e) {
+            if (!persist) {
+              appActions.removeSiteSetting(origin, 'autoplay')
+              // tab.removeListener('did-finish-load', temporaryAllow)
```
Comment (bbondy) [inline app/browser/reducers/autoplayReducer.js]: nit: pls remove this commented out code

**Comment by bbondy** [comment]
++

**Pull request merged**